CMakeLists.txt

```diff
@@ -25,8 +33,38 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     # TODO: A better fix should handle ld's --as-needed flag 
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Xlinker '--no-as-needed'")
 elseif(MSVC)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DNOMINMAX") # exclude M$ min/max macros
     #set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /analyze")
+    add_definitions(-DNOMINMAX) # exclude M$ min/max macros
+    add_definitions(-D_SCL_SECURE_NO_WARNINGS) #turn off msvc warning
+    add_definitions(-D_WINSOCK_DEPRECATED_NO_WARNINGS) #likewise
+    if(APPVEYOR)
+        link_directories("C:/Libraries/boost/lib32-msvc-12.0")
+        add_definitions(-D_WIN32_WINNT=0x0600) #windows vista
+        message(STATUS "MSVC -> forcing use of statically-linked runtime.")
+        set(variables
+          CMAKE_C_FLAGS_DEBUG
+          CMAKE_C_FLAGS_MINSIZEREL
+          CMAKE_C_FLAGS_RELEASE
+          CMAKE_C_FLAGS_RELWITHDEBINFO
+          CMAKE_CXX_FLAGS_DEBUG
+          CMAKE_CXX_FLAGS_MINSIZEREL
+          CMAKE_CXX_FLAGS_RELEASE
+          CMAKE_CXX_FLAGS_RELWITHDEBINFO
+        )
+        if(verbose)
+        message(STATUS "Initial build flags:")
+        show(variables)
+        endif()
+        foreach(variable ${variables})
+            if(${variable} MATCHES "/MD")
+                string(REGEX REPLACE "/MD" "/MT" ${variable} "${${variable}}")
+            endif()
+        endforeach()
+        if(verbose)
+            message(STATUS "Build flags after replace:")
+            show(variables)
+        endif()
+    endif()
```

Comment:
Please don't copy/paste code from StackOverflow. Why is a static runtime needed? Isn't it in the library path?